[FIX] payment_stripe: add the documentation for local payment methods (#774)
odoo/documentation-user: general/payment_acquirers/stripe.rst

======
Stripe
======

`Stripe <https://stripe.com/>`__ is a United States-based online payment solution
provider, allowing businesses to accept **credit cards** and other payment methods.

Local Payment Methods
=====================

> `Local payment methods <https://stripe.com/payments/payment-methods-guide>`__ are payment methods which are

Author: done

only available for certain merchants and customers countries and currencies.

> Odoo supports the integration of the following local payment methods:

Author: done

- Bancontact
- EPS
- Giropay
- iDeal:
- Przelewy24 (P24)

> In order to support one local payment method, you must add its icon to the supported payment method icons:
> Payment Acquirers > Stripe > Configuration > Supported Payment Icons.

Reviewer: ```suggestion
To offer a given local payment method to your customers, it must be listed as a supported payment icon in Stripe. Go to :menuselection:`Payment Acquirers --> Stripe --> Configuration` and add the desired payment method in **Supported Payment Icons**.
```
@jcs-odoo Could you please check this writing?

@AdrienHorgnies 
- https://www.odoo.com/documentation/user/14.0/contributing/documentation/rst_guidelines.html#use-the-menuselection-directive
- This could benefit from an illustration (don't forget to use *pngquant* to compress it before committing).

Author: done

Reviewer 2: sure, I post my suggestion as a different review. :)

> If the icon of a payment method doesn't exist at all, **Odoo will offer the corresponding payment method by default**.
> In order to disable such a payment method, update the Stripe module to get the new icons or create the icon yourself and
> don't add it to the supported payment icons.

Reviewer: ```suggestion
.. note::
   If a payment icon does not exist at all in the database, the corresponding local payment method
   is always offered to customers.
```
I wouldn't recommend users to create the icon by themselves since they must figure out the exact name to use, and I would certainly not recommend them to upgrade their `payment` module just for this.
Instead, let's just stick to the description of the behavior and we'll see later is something more is required.

Author: How will the customers get the icon I added again ? I thought they had to update payment_stripe to get them. Well, I guess the information will be part of the support mail we'll send, so indeed it doens't need to be here.

Reviewer: The icons are defined in the `payment` module, not in `payment_stripe`. Upgrading such a module is not a trivial action so we shouldn't rely on that to ensure a working fix.

======
Stripe
======

`Stripe <https://stripe.com/>`_ is a United States-based online payment solution
provider, allowing businesses to accept **credit cards** and other payment methods.

Enable Local Payment Methods
============================

Local payment methods are payment methods that are only available for certain merchants
and customers countries and currencies.

> The Stripe connector in Odoo supports the integration of the following local payment methods:

Reviewer: ```suggestion
The Stripe connector in Odoo supports the following local payment methods:
```
We don't integrate payment methods.

Author: Diff applied

> The Stripe connector in Odoo supports the integration of the following local payment methods:
> 
> - Bancontact
> - EPS
> - Giropay
> - iDeal:
> - Przelewy24 (P24)

Reviewer 2: Is this list exhaustive?
Your screenshot suggests that there are more payment methods. If this relies on Stripe, I would advise you to link the text "local payment methods" with the page listing all methods available and rephrase your sentence.

```suggestion
The Stripe connector in Odoo supports many local payment methods, including:

- Bancontact
- EPS
- Giropay
- iDeal:
- Przelewy24 (P24)
```

Reviewer: The list is indeed exhaustive. Other payment methods are not global.

> To offer a given local payment method to your customers, it must be listed as a supported payment icon in Stripe.
> Go to :menuselection:`Payment Acquirers --> Stripe --> Configuration` and add the desired payment method in
> **Supported Payment Icons**.

Reviewer 2: Suggestion:
```suggestion
To enable specific Local Payment Methods with Stripe, list them as supported payment icons. To do so, go
to :menuselection:`Payment Acquirers --> Stripe --> Configuration` and add the payment methods in the
**Supported Payment Icons** field.
```

Reviewer: Ok for me but then add "If the desired payment method is already listed, you don't have anything to do.".

The reason is that this page's first purpose is to guide users after the deployment of a bugfix that adds this behavior. Most users won't have anything to do so we shouldn't let them think that they need to change anything when they actually don't.

Author: > To enable specific Local Payment Methods with Stripe, list them as supported payment icons.
To do so, go to :menuselection:`Payment Acquirers --> Stripe --> Configuration` and add the desired payment methods in
the **Supported Payment Icons** field. If the desired payment method is already listed, you don't have anything to do.

> .. image:: media/stripe_enable_local_payment_method.png

Reviewer 2: Screenshots are preferably taken on enterprise databases, if possible.

Author: I replaced screenshot with same content but from enterprise.

   :align: center
   :alt: Select and add icons of payment methods you want to enable

.. note::

> If a payment method icon doesn't exist at all in the database, the corresponding local payment method is always offered
> to customers.

Reviewer: ```suggestion
   If a payment method icon doesn't exist at all in the database, the corresponding local payment method is always offered
   to customers.
```
The note doesn't show unless its content is properly indented.

Author: diff applied
